mikeio1d/result_network/result_quantity.py

```diff
@@ -73,12 +96,21 @@ def to_txt(self, file_path, time_step_skipping_number=1):
 
     def get_query(self):
         """Get query corresponding to ResultQuantity."""
-        return self.result_location.get_query(self.data_item)
+        return self._timeseries_id.to_query()
 
     def get_data_entry(self):
         """Get DataEntry corresponding to ResultQuantity."""
-        return DataEntry(self.data_item, self.element_index)
+        return DataEntry(self.data_item, self.element_index, self.m1d_dataset)
 
     def get_data_entry_net(self):
         """Get DataEntryNet corresponding to ResultQuantity."""
         return DataEntryNet(self.data_item, self.element_index)
+
+    @property
+    def timeseries_id(self) -> TimeSeriesId:
+        """TimeSeriesId corresponding to ResultQuantity."""
+        if self._timeseries_id is None:
+            raise ValueError(
```

Comment:
Use `message = "ResultQuantity must be added to a ResultNetwork before TimeSeriesId can be accessed."` and then `raise ValueError(message)`?